kratos/spatial_containers/spatial_search_result.h

```diff
@@ -143,6 +144,7 @@ class SpatialSearchResult
         if constexpr (std::is_base_of_v<IndexedObject, TObjectType>) {
             return mpObject->Id();
         } else { // Otherwise 0
+            KRATOS_WARNING("SpatialSearchResult") << "Object does not provide an ID" << std::endl;
```

Comment:
what does this mean? I.e. what should I do as user?

I feel like if some object (i.e. objects without IDs) should not be allowed to be used if they anyway wont work

Reply:
Do you prefer an error, I have doubts about it

Reply:
ideally an error where this `SpatialSearchResult` is used, at construction time

Reply:
The Id it is a new method, I would add the error only when calling the Id method, not everytime the class is constructed, because may not be used at all

Reply:
Replaced